hw/mcu/nordic/nrf52xxx/src/hal_gpio.c

```diff
@@ -139,6 +139,27 @@ hal_gpio_init_out(int pin, int val)
     return 0;
 }
 
+/**
+ * Deinitialize the specified pin to revert the previous initialization
+ *
+ * @param pin Pin number to unset
+ *
+ * @return int  0: no error; -1 otherwise.
+ */
+int
+hal_gpio_deinit(int pin)
+{
+    uint32_t conf;
+    NRF_GPIO_Type *port;
+
+    conf = GPIO_PIN_CNF_INPUT_Disconnect << GPIO_PIN_CNF_INPUT_Pos;
+    port = HAL_GPIO_PORT(pin);
+    port->PIN_CNF[pin] = conf;
```

Comment:
you need to use `HAL_GPIO_INDEX` here for this to work properly on nRF52840

Reply:
Correct, fixed by: https://github.com/apache/mynewt-core/pull/1630/commits/36004f1985f4d34758cf5204a84ea29adfc0fae6